classes/Product.php

```
@@ -7113,10 +7121,42 @@ public function getWsStockAvailables()
     }
 
     /**
-     * Webservice getter : Get Tag identifiers
+     * Webservice getter : Get product attachments ids of current product for association
      *
      * @return array
      */
+    public function getWsAttachments()
+    {
+        $result = Db::getInstance()->executeS('SELECT a.`id_attachment` AS id
+			FROM `' . _DB_PREFIX_ . 'product_attachment` pa
+			LEFT JOIN `' . _DB_PREFIX_ . 'attachment` a ON (pa.id_attachment = a.id_attachment)
+			' . Shop::addSqlAssociation('attachment', 'a') . '
+			WHERE pa.`id_product` = ' . (int) $this->id);
+
+        return $result;
+    }
+
+    /**
+     * Webservice setter : set product attachments ids of current product for association
+     *
+     * @param $attachments ids
+     */
+    public function setWsAttachments($attachments)
+    {
+        try {
+            $this->deleteAttachments(true);
+            foreach ($attachments as $attachment) {
+                Db::getInstance()->execute('INSERT INTO `' . _DB_PREFIX_ . 'product_attachment`
+    				(`id_product`, `id_attachment`) VALUES (' . (int) $this->id . ', ' . (int) $attachment['id'] . ')');
+            }
+            Product::updateCacheAttachment((int) $this->id);
+        } catch (Exception $e) {
+            return false;
+        }
```

Again, bad idea to hide exceptions

---

I agree